File: Lecture 10: Solving LeetCode Arrays/pairSum.java
import java.util.*;

public class Solution{
    public static int[][] pairSum(int[] arr, int s) {
        // Write your code here.
        
        List< List<Integer> > ans = new ArrayList<>();
   		
        Arrays.sort(arr);
        
       for(int i=0;i<arr.length; i++ )
       {
            for(int j = i+1; j<arr.length; j++) {
                if(arr[i] +arr[j] == s)
                {
                    List<Integer> temp = new ArrayList<>();
                    temp.add(Math.min(arr[i], arr[j]));
                    temp.add(Math.max(arr[i], arr[j]));
                    ans.add(temp);
                }
            }
       }
        	int[][] pairs = new int[ans.size()][2];
        	
        	int k = 0;
        
        	for(List<Integer> num:ans){
                pairs[k][0] = (int)num.get(0);
                pairs[k][1] = (int)num.get(1);
                k++;
            }
   			return pairs;
    }
}

File: 010_Solving_LeetCode_Arrays/pairSum.cpp
#include<vector>
using namespace std;
vector<vector<int> > pairSum(vector<int> &arr, int s){
   vector< vector<int> > ans;
   
   for(int i=0;i<arr.size(); i++ )
   {
   		for(int j = i+1; j<arr.size(); j++) {
        	if(arr[i] +arr[j] == s)
            {
            	vector<int> temp;
                temp.push_back(min(arr[i], arr[j]));
                temp.push_back(max(arr[i], arr[j]));
				ans.push_back(temp);
            }
        }
   }
   sort(ans.begin(), ans.end());
   return ans;
}